### ISSUE TITLE
Rule Proposal: spread-spacing

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#5391]
I'd like to enforce spacing between the spread operator (`...`) and its expression.  I propose calling it `spread-spacing`

**When does this rule warn? Please describe and show example code:**

I envision it being configured with strings "always" and "never". 

With "never" (the default):

```js
// Warns
myFunction(... args);

// Does not warn
myFunction(...args);
```

And vice-versa for "always"

**Is this rule preventing an error or is it stylistic?**

Stylistic

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

It seems to fit with the other spacing rules we've included in core.

**Are you willing to create the rule yourself?**

Yes.

**Comment by BYK** [comment]
:+1:

**Comment by nzakas** [comment]
Is this just function call spread? What about array spread?

```js
x = [...y];
```

And what about rest?

```js
[x, ...y] = foo;
```

**Comment by IanVS** [comment]
I propose this would apply to all forms of spread operators.  If people want more configurability in the future, it could be expanded with further options for the various cases.  But, I don't see much of a use case for enforcing/prohibiting spaces for certain kinds and not others (although I'm sure someone will want to eventually).

**Comment by mysticatea** [comment]
Sounds good to me :+1:

**Comment by nzakas** [comment]
Sounds good

**Comment by 3den** [comment]
the spread operator is not working working for me on version 2.8.0:

```yaml
extends: "eslint:recommended"

env:
  browser: true
  es6: true
  jasmine: true

parserOptions:
  ecmaVersion: 6
  ecmaFeatures:
    jsx: true
    spread: true
```

**Comment by nzakas** [comment]
@3den your comment isn't related to this issue, please stop by https://gitter.im/eslint/eslint if you need help configuring ESLint.

**Comment by kaicataldo** [comment]
Are you working on this, @IanVS? If not, I'd be happy to take it on :)

**Comment by nzakas** [comment]
@kaicataldo you've volunteered for a bunch of issues, could I entice you to spend some of that energy on the JSCS compat tasks? :)

**Comment by kaicataldo** [comment]
@nzakas Sure - I've been going through old issues to see if I can clean some of them up, but understand the need to focus on higher priority things.

**Comment by nzakas** [comment]
@kaicataldo much appreciated!
